### ISSUE TITLE
NvInferRuntimeCommon.h is #included in sample code

### ISSUE THREAD

**Issue by carlosgalvezp** [NVIDIA/TensorRT#3363]
Hi,

`NvInferRuntimeCommon.h` [clearly states](https://github.com/NVIDIA/TensorRT/blob/release/8.6/include/NvInferRuntimeCommon.h#L24) that it must not be included:

```cpp
//! \warning Do not directly include this file. Instead include NvInferRuntime.h
```

However, it's included in many places in the `samples` directory, for example [here](https://github.com/NVIDIA/TensorRT/blob/release/8.6/samples/common/ErrorRecorder.h#L20).

I take it that `NvInferRuntimeCommon.h` is an implementation detail and thus we clients should not include it directly. Could therefore the `samples` be updated to reflect best practices?

Thanks!

**Comment by zerollzeng** [comment]
@rajeevsrao ^ ^

**Comment by ttyio** [comment]
@zerollzeng could you file a bug to track this? thanks!

**Comment by zerollzeng** [comment]
Filed internal bug 4411864 to track

**Comment by ttyio** [comment]
this is fixed in rel-10.0, see https://github.com/NVIDIA/TensorRT/blob/release/10.0/samples/common/ErrorRecorder.h,

closing and thanks all!
